<launch>
<group ns="nanoplus">

<node pkg="nodelet"
	type="nodelet"
	name="nodelet_manager"
	args="manager"/>

<node pkg="quad_controller"
  type="quad_trajectory"
  name="quad_trajectory"
  output="screen">
  <param name="des_pos/x" value="0"/>
  <param name="des_pos/y" value="0"/>
  <param name="des_pos/z" value="0"/>
  <param name="time/scale" value="1"/>
  <param name="time/takeoff" value="4"/>
  <param name="time/goto" value="2"/>
  <param name="time/land" value="1500"/>
  <remap from="~trajectory" to="trajectory"/>
  <remap from="~trajectoryinfo" to="trajectoryinfo"/>
  <remap from="~odom" to="/odom"/>
  <remap from="~startsignal" to="/Killswitch/Killsignal"/>
  </node>

<!--
  <node pkg="quad_controller"
  type="quad_trajectory"
  name="quad_trajectory2"
  output="screen">
  <param name="des_pos/x" value="0.2"/>
  <param name="des_pos/y" value="0.2"/>
  <param name="des_pos/z" value="1.2"/>
  <param name="time/scale" value="1"/>
  <param name="time/takeoff" value="4"/>
  <param name="time/goto" value="2"/>
  <param name="time/land" value="1500"/>
  <remap from="~trajectory" to="trajectory2"/>
  <remap from="~trajectoryinfo" to="trajectoryinfo2"/>
  <remap from="~odom" to="/odom2"/>
  <remap from="~startsignal" to="/Killswitch/Killsignal"/>
  </node>

  <node pkg="quad_controller"
  type="quad_trajectory"
  name="quad_trajectory3"
  output="screen">
  <param name="des_pos/x" value="0.2"/>
  <param name="des_pos/y" value="-0.2"/>
  <param name="des_pos/z" value="1.2"/>
  <param name="time/scale" value="1"/>
  <param name="time/takeoff" value="4"/>
  <param name="time/goto" value="2"/>
  <param name="time/land" value="1500"/>
  <remap from="~trajectory" to="trajectory3"/>
  <remap from="~trajectoryinfo" to="trajectoryinfo3"/>
  <remap from="~odom" to="/odom3"/>
  <remap from="~startsignal" to="/Killswitch/Killsignal"/>
  </node>

  <node pkg="quad_controller"
  type="quad_trajectory"
  name="quad_trajectory4"
  output="screen">
  <param name="des_pos/x" value="-0.2"/>
  <param name="des_pos/y" value="-0.2"/>
  <param name="des_pos/z" value="1.2"/>
  <param name="time/scale" value="1"/>
  <param name="time/takeoff" value="4"/>
  <param name="time/goto" value="2"/>
  <param name="time/land" value="1500"/>
  <remap from="~trajectory" to="trajectory4"/>
  <remap from="~trajectoryinfo" to="trajectoryinfo4"/>
  <remap from="~odom" to="/odom4"/>
  <remap from="~startsignal" to="/Killswitch/Killsignal"/>
  </node>
-->

  <node pkg="quad_controller"
  type="gen_trajectory"
  name="gen_trajectory"
  output="screen">
  <remap from="~trajectoryinfo" to="trajectoryinfo"/>
  <remap from="~trajectoryinfo2" to="trajectoryinfo2"/>
  <remap from="~trajectoryinfo3" to="trajectoryinfo3"/>
  <remap from="~trajectoryinfo4" to="trajectoryinfo4"/>
  </node>

<node pkg="quad_controller"
type="test_control"
name="test_control"
output="screen">
<remap from="~trpy_cmd" to="trpy_cmd"/>
<remap from="~odom" to="/odom"/>
<remap from="~trajectory" to="trajectory"/>
<remap from="~euler" to="euler"/>
<remap from="~killsignal" to="/Killswitch/Killsignal"/>
<param name="output/pos" value="pos1.txt"/>
<param name="output/vel" value="vel1.txt"/>
</node>

<!--
<node pkg="quad_controller"
type="test_control"
name="test_control2"
output="screen">
<remap from="~trpy_cmd" to="trpy_cmd2"/>
<remap from="~odom" to="/odom2"/>
<remap from="~trajectory" to="trajectory2"/>
<remap from="~euler" to="euler2"/>
<remap from="~killsignal" to="/Killswitch/Killsignal"/>
<param name="output/pos" value="pos2.txt"/>
<param name="output/vel" value="vel2.txt"/>
</node>

<node pkg="quad_controller"
type="test_control"
name="test_control3"
output="screen">
<remap from="~trpy_cmd" to="trpy_cmd3"/>
<remap from="~odom" to="/odom3"/>
<remap from="~trajectory" to="trajectory3"/>
<remap from="~euler" to="euler3"/>
<remap from="~killsignal" to="/Killswitch/Killsignal"/>
<param name="output/pos" value="pos3.txt"/>
<param name="output/vel" value="vel3.txt"/>
</node>

<node pkg="quad_controller"
type="test_control"
name="test_control4"
output="screen">
<remap from="~trpy_cmd" to="trpy_cmd4"/>
<remap from="~odom" to="/odom4"/>
<remap from="~trajectory" to="trajectory4"/>
<remap from="~euler" to="euler4"/>
<remap from="~killsignal" to="/Killswitch/Killsignal"/>
<param name="output/pos" value="pos4.txt"/>
<param name="output/vel" value="vel4.txt"/>
</node>
-->
</group>
</launch>
